hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -219,10 +220,22 @@ protected HoodieMetadataPayload(String key, int type,
                                                                                Option<Map<String, Long>> filesAdded,
                                                                                Option<List<String>> filesDeleted) {
     Map<String, HoodieMetadataFileInfo> fileInfo = new HashMap<>();
-    filesAdded.ifPresent(
-        m -> m.forEach((filename, size) -> fileInfo.put(filename, new HoodieMetadataFileInfo(size, false))));
-    filesDeleted.ifPresent(
-        m -> m.forEach(filename -> fileInfo.put(filename, new HoodieMetadataFileInfo(0L, true))));
+    filesAdded.ifPresent(filesMap ->
+        fileInfo.putAll(
+            filesMap.entrySet().stream().collect(
+                Collectors.toMap(Map.Entry::getKey, (entry) -> {
+                  long fileSize = entry.getValue();
+                  // Assert that the file-size of the file being added is positive, since Hudi
+                  // should not be creating empty files
+                  checkState(fileSize > 0);
+                  return new HoodieMetadataFileInfo(fileSize, false);
+                })))
+    );
+    filesDeleted.ifPresent(filesList ->
+        fileInfo.putAll(
```

Comment:
minor: the forEach version is much simpler and clearer

Reply:
Agreed. Please check my previous comment regarding GC impact